CoreFeatures/Base/SynchronizedLazyNamedItemsStorageStrategy/src/main/java/info/smart_tools/smartactors/base/synchronized_lazy_named_items_storage_strategy/SynchronizedLazyNamedItemsStorageStrategy.java

```diff
@@ -0,0 +1,130 @@
+package info.smart_tools.smartactors.base.synchronized_lazy_named_items_storage_strategy;
+
+import info.smart_tools.smartactors.base.interfaces.i_addition_dependency_strategy.IAdditionDependencyStrategy;
+import info.smart_tools.smartactors.base.interfaces.i_addition_dependency_strategy.exception.AdditionDependencyStrategyException;
+import info.smart_tools.smartactors.base.interfaces.iresolve_dependency_strategy.IResolveDependencyStrategy;
+import info.smart_tools.smartactors.base.interfaces.iresolve_dependency_strategy.exception.ResolveDependencyStrategyException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Strategy that stores named objects resolved lazily using added strategies.
+ *
+ * <p>
+ *  A strategy for a object name may be changed or removed until the object for that name is resolved.
+ * </p>
+ *
+ * <p>
+ *  All operations are synchronized.
+ * </p>
+ */
+public class SynchronizedLazyNamedItemsStorageStrategy
+        implements IResolveDependencyStrategy, IAdditionDependencyStrategy {
+    /**
+     * A object that lazily resolves a single item.
+     */
+    private static final class LazyRef {
```

Comment:
Why not?

The class `LazyRef` is a implementation detail of `SynchronizedLazyNamedItemsStorageStrategy` and is not a part of it's interface so defining it as an inner class seems to be more convenient. It also provides a bit better access control.

reasone->reason

Reply:
I am very strongly against inner classes because
- they violate a java conception (one file - one class);
- they add complexity to the source file;
- they add complexity to debugging and profiling.

Reply:
* one file-one class. In this case the inner class is a implementation detail of the outer class and is used only by the outer class. The same behaviour could be implemented in outer class methods. So conceptually the inner class is just a part of outer class. In the end there are inner classes in Java. In the end there is only one *top-level class* in file.
* source file complexity. Isn't it more complex to read two files to understand meaning of one of them when another one is not used by any other?
* debugging/profiling complexity. I really need explaination/example of how inner classes influence debugging and profiling as I didn't ever have any such problems.